### PR TITLE
Add styled-components with typescript example

### PR DESCRIPTION
**Pull request by jorgeortega**
Adding a simple example with styled-components and TypeScript

### CODE REVIEW

File: examples/styled-components/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import styled from 'styled-components'
+
+interface ButtonProps extends HTMLButtonElement {
+	btnType: 'primary' | 'secondary',
```
Comment (sapegin) [inline examples/styled-components/src/components/Button/Button.tsx]: Let's simplify this a bit: rename a prop to `variant` and pass it to a styled component, so we don't duplicate the whole component markup twice.

Or just copy code here: 
https://github.com/component-driven/component-driven-development/blob/master/src/components/core/Button/Button.js

File: examples/styled-components/styleguide.config.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+				}
+			],
+			noParse: /\.(css|scss)/
+		},
+		resolve: {
+			modules: [
+			  path.resolve(__dirname, 'src'),
+			  path.join(__dirname, 'node_modules')
+			],
+			extensions: ['.js', '.ts', '.tsx']
```
Comment (sapegin) [inline examples/styled-components/styleguide.config.js]: Make sense to update the default extensions list:
https://github.com/styleguidist/react-styleguidist/blob/0cd533c18a3ac79bfec2c0e34f65780a748bea0a/src/scripts/make-webpack-config.js#L44

File: examples/styled-components/styleguide.config.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+				{
+					test: /\.(js|ts)x?$/,
+					exclude: /node_modules/,
+					loader: 'babel-loader',
+				}
+			],
+			noParse: /\.(css|scss)/
+		},
+		resolve: {
+			modules: [
```
Comment (sapegin) [inline examples/styled-components/styleguide.config.js]: Is this necessary?

File: examples/styled-components/Readme.md [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# React Styleguidist styled-components typescript example style guide
```
Comment (sapegin) [inline examples/styled-components/Readme.md]: It's TypeScript.

File: examples/styled-components/Readme.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+See [deployed version](https://react-styleguidist.js.org/examples/styled-components/).
```
Comment (sapegin) [inline examples/styled-components/Readme.md]: This won't work, we only deploy basic example. But maybe we should deploy other examples too…

**Comment by jorgeortega** [comment]
@sapegin Thanks for your feedback. All changes updated.

**Comment by sapegin** [comment]
Thanks!

**Pull request merged**